--- easy_e17.sh	2012-04-13 19:29:10.919892145 +0800
+++ easy_e17.sh.new02	2012-04-13 19:31:13.443895908 +0800
@@ -149,10 +149,6 @@
 					echo "      --srcrev=<revision>             = set the default source revision"
 					echo "      --asuser                        = do everything as the user, not as root"
 					echo "      --no-sudopwd                    = sudo don't need a password..."
-					echo "  -c, --clean                         = clean the sources before building"
-					echo "                                        (more --cleans means more cleaning, up"
-					echo "                                        to a maximum of three, which will"
-					echo "                                        uninstall e17)"
 					echo "  -s, --skip-srcupdate                = don't update sources"
 					echo "  -a, --ask-on-src-conflicts          = ask what to do with a conflicting"
 					echo "                                        source file"
@@ -510,39 +506,6 @@
 	rm -f "$status_path/$name.noerrors"
 	rm -f "$logs_path/$name.log"
 
-	if [ $clean -ge 1 ]; then
-		if [ -e "Makefile" ]; then
-			if [ $clean -eq 1 ]; then
-				run_command "$name" "$path" "clean" "clean  : " "$mode" "$make -j $threads clean"
-				if [ ! -e "$status_path/$name.noerrors" ]; then
-					if [ "$skip_errors" ]; then
-						write_appname "$name" "hidden"	# clean might fail, that's ok
-					else
-						return
-					fi
-				fi
-			fi
-			if [ $clean -eq 2 ]; then
-				run_command "$name" "$path" "distclean" "distcln: " "$mode" "$make -j $threads clean distclean"
-				if [ ! -e "$status_path/$name.noerrors" ]; then
-					if [ "$skip_errors" ]; then
-						write_appname "$name" "hidden"	# distclean might fail, that's ok
-					else
-						return
-					fi
-				fi
-			fi
-			if [ $clean -ge 3 ]; then
-				run_command "$name" "$path" "uninstall" "uninst : " "rootonly" "$make -j $threads uninstall clean distclean"
-				if [ ! -e "$status_path/$name.noerrors" ] ; then return ; fi
-
-			    # It's no longer installed if we just uninstalled it.
-			    # Even if the uninstall failed, it's best to mark it as uninstalled so that a partial uninstall gets fixed later.
-			    rm -f $status_path/$name.installed
-			fi
-		fi
-	fi
-
 	# get autogen arguments
 	args=""
 	for app_arg in `echo $autogen_args | tr -s '\,' ' '`; do
@@ -855,7 +818,6 @@
 accache=""
 easy_options=""
 command_options=$@
-clean=0
 mkdir -p "$HOME/.config/easy_e17" || {
     echo >&2 "ERROR: Can not create config directory $HOME/.config/easy_e17"
     exit 1
@@ -962,7 +924,6 @@
 			;;
 		--asuser)					asuser=1 ;;
 		--no-sudopwd)				no_sudopwd=1 ;;
-		-c|--clean)					clean=$(($clean + 1))	;;
 		-d|--docs)					gen_docs=1 ;;
 		--postscript)				easy_e17_post_script="$value" ;;
 		-s|--skip-srcupdate)		skip_srcupdate=1 ;;
