--- easy_e17.sh	2012-04-13 19:33:38.739900371 +0800
+++ easy_e17.sh.new04	2012-04-13 19:39:27.747911088 +0800
@@ -520,6 +520,9 @@
 		fi
 	done
 	
+	if [ -e "$HOME/.config/easy_e17/$name.patch" ] ; then
+		run_command "$name" "$path" "patch"   "patch:	" "$mode"    "patch -p1 -i $HOME/.config/easy_e17/$name.patch"
+	fi
 	if [ -e "autogen.sh" ]; then
 		run_command "$name" "$path" "autogen" "autogen: " "$mode"    "./autogen.sh --prefix=$install_path $accache $args"
 		if [ ! -e "$status_path/$name.noerrors" ] ; then return ; fi
